compiler-core/src/javascript.rs

```diff
@@ -317,33 +317,27 @@ impl<'a> Generator<'a> {
             .iter()
             .map(|arg| {
                 arg.label.as_ref().map(|(_, name)| {
-                    docvec![
-                        "'",
-                        name,
-                        "' in fields ? fields.",
-                        name,
-                        " : this.",
-                        name,
-                        ","
-                    ]
+                    docvec!["'", name, "' in fields ? fields.", name, " : this.", name,]
                 })
             })
             .collect::<Option<Vec<_>>>();
 
-        if let Some(with_fields_constructor_args) = with_fields_constructor_args {
-            definitions.push(docvec![
-                "withFields(fields) {",
-                docvec![
-                    line(),
-                    "return new this.constructor(",
-                    docvec![line(), join(with_fields_constructor_args, line())].nest(INDENT),
+        if let Some(args) = with_fields_constructor_args {
+            if args.len() > 1 {
+                definitions.push(docvec![
+                    "withFields(fields) {",
+                    docvec![
+                        line(),
+                        "return new this.constructor(",
```

Comment:
It *might* further help some JS optimisers to have the exact name specified here, i.e. insert the value of `constructor.name`, rather than doing a runtime lookup of `this.constructor`.

Reply:
It might also be slower, since it replaces the same property lookup on `this` with a different one on `globalThis` for each type. I couldn't measure a difference. But I think it would generate more idiomatic code, so it's most likely not worse, and it will definitely be more readable.